src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
+        if (index == 0) {
+            if (head == null) {
+                tail = head = newNode;
+            } else {
+                newNode.next = head;
+                head.prev = newNode;
+                head = newNode;
+            }
+            size++;
+        } else if (index == size) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        } else {
+            Node<T> current = head;
```

Comment:
Consider creating the `findNodeByIndex` method to avoid code duplication in methods like `add(T value, int index)`, `get(int index)`, `set(T value, int index)`, and `remove(int index)`.